src/content/1.7/development/components/grid/tutorials/work-with-search-form/_index.md

```diff
@@ -0,0 +1,293 @@
+
+- `GridDefinition::getFilters`
+- `Filters` class
+- Controllers:
+  - Search action
+  - List action
+  - Reset filter action
+
+## Add Filters into GridDefinition
+
+You should use `AbstractGridDefinitionFactory` as a parent class, and define a const `GRID_ID` which will be used as a key to persist the `Filters`.
+
+```php
+final class ManufacturerGridDefinitionFactory extends AbstractGridDefinitionFactory {
+    const GRID_ID = 'manufacturer';
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getId()
+    {
+        return self::GRID_ID;
+    }
+
+    ...
+    /**
+    * {@inheritdoc}
+    */
+    protected function getFilters()
+    {
+        return (new FilterCollection())
+            ->add((new Filter('id_manufacturer', TextType::class))
+                ->setTypeOptions([
+                    'required' => false,
+                    'attr' => [
+                        'placeholder' => $this->trans('Search ID', [], 'Admin.Actions'),
+                    ],
+                ])
+                ->setAssociatedColumn('id_manufacturer')
+            )
+            ->add((new Filter('name', TextType::class))
+                ->setTypeOptions([
+                    'required' => false,
+                    'attr' => [
+                        'placeholder' => $this->trans('Search name', [], 'Admin.Actions'),
+                    ],
+                ])
+                ->setAssociatedColumn('name')
+            )
+            ->add((new Filter('active', YesAndNoChoiceType::class))
+                ->setAssociatedColumn('active')
+            )
+            ->add((new Filter('actions', SearchAndResetType::class))
+                ->setAssociatedColumn('actions')
+                ->setTypeOptions([
+                    'reset_route' => 'admin_common_reset_search_by_filter_id',
+                    'reset_route_params' => [
+                        'filterId' => self::GRID_ID,
+                    ],
+                    'redirect_route' => 'admin_manufacturers_index',
+                ])
+                ->setAssociatedColumn('actions')
+            )
+        ;
+    }
+    ...
+}
+```
+
+### The filters types
+
+In the filters collection you define all the available filters (which will match your grid columns), you can define a specific type depending on the column.
+You can basically use any Symfony form type (including your custom ones) and PrestaShop provides a few [filter types]({{< ref "/1.7/development/components/grid/filter-types-reference/_index.md" >}}) that might be useful to you.
+
+### Filterable grid definition
+{{< minver v="1.7.7" title="true" >}}
+
+Starting the 1.7.7 we made it even easier, you can use the `AbstractFilterableGridDefinitionFactory`. This will allow you use the [common search controller](#common-search-controller).
+
+```php
+final class AddressGridDefinitionFactory extends AbstractFilterableGridDefinitionFactory {
```

Comment:
```suggestion
final class ManufacturerGridDefinitionFactory extends AbstractFilterableGridDefinitionFactory {
```

That permits to have a sample who follow the page.

Reply:
This is on purpose, because we don't explain the same way: `AbstractFilterableGridDefinitionFactory` vs `AbstractGridDefinitionFactory` which is only possible in 177

And it matches the current grid definitions in the core which are also different

Reply:
Ok. Misread